### PR TITLE
fix(discord): push connected status when gateway already connected at lifecycle start

### PR DESCRIPTION
**Pull request by mitchmcalister**
## Summary

- When `runDiscordGatewayLifecycle` starts, it registers a debug event listener to track `"WebSocket connection opened"` events and push `connected: true` to the channel status sink.
- If the gateway completes its READY handshake *before* the lifecycle registers the listener (a race window of ~30ms observed in production), the initial event is missed and `connected` stays `undefined`.
- The health monitor (`channel-health-monitor.ts`) treats `connected === false` (or undefined, which is falsy) as unhealthy and triggers a restart every check cycle, creating a restart loop.
- This adds a `gateway.isConnected` check immediately after listener registration to catch the already-connected case.

## Reproduction

1. Discord gateway connects and emits READY before `runDiscordGatewayLifecycle` registers its debug listener
2. `connected` is never set to `true` in channel runtime
3. Health monitor detects `running: true, connected: false` → restarts channel as "stuck"
4. Restart reconnects, same race happens again → infinite restart loop every ~10 minutes

## Test plan

- [x] Added test: `pushes connected status when gateway is already connected at lifecycle start`
- [x] Existing lifecycle tests pass (10/10)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/discord/monitor/provider.lifecycle.ts [hunk 1]
```diff
@@ -244,6 +244,19 @@ export async function runDiscordGatewayLifecycle(params: {
...
+  // registered the listener above), push the initial connected status now.
+  if (gateway?.isConnected) {
+    const at = Date.now();
+    pushStatus({
+      connected: true,
+      lastEventAt: at,
+      lastConnectedAt: at,
+      lastDisconnect: null,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (mitchmcalister) [reply src/discord/monitor/provider.lifecycle.ts]: Thanks! Addressed in 96c6011 — added `!lifecycleStopping` guard to the `isConnected` check so a pre-aborted signal's `connected: false` isn't contradicted, plus a test for the edge case.
